BLSQ/iaso
# Ia 2068 json data store

In order to receive data calculated outside of Iaso, we need an endpoint that lets users define a slug and associates json data to it

Related JIRA tickets : IA-2068

## Self proofreading checklist


## Changes

- Add `/api/datastore/` endpoint. The endpoint supports GET, POST, PUT and DELETE
- Add separate read and write permissions for the endpoint. The permissions are not visible from the front-end but can be attributed through the Django admin
- Add JsonDataStore entry to the Django admin
- GET: results returned are account specific (you only see data stores for the account of the user you're logged as)
- POST: the body must include `key` (the slug) and `data` (the json). The key must be unique for the account
- PUT: must include all fields. Will return a 400 if you try to update a key with another that already exists for your account.
- DELETE: well, it deletes.

## How to test

- go to /api/datastore/
- try creating a few stores by POSTing
- try to access a specific store directly via `/api/datastore/{slug}`
- try to update a store with PUT
- try to DELETE and see if the list updates

## Print screen / video

https://user-images.githubusercontent.com/38907762/233620678-76d8dcfd-abb8-4a8d-99d0-f35273fa22b1.mov

## Notes

I used `slugify` to sanitize the slugs, I don't know if it enough
